megaavr/libraries/Logic/src/Logic.cpp

```diff
@@ -81,22 +81,26 @@ void CustomLogic::init(block_t &block)
 
 void CustomLogic::attachInterrupt(block_t &block, void (*userFunc)(void), PinStatus mode)
 {
+  CCL_INTMODE0_t intmode;
   switch (mode) 
   {
     // Set RISING, FALLING or CHANGE interrupt trigger for a block output
     case RISING:
-      CCL.INTCTRL0 |= (0x03 << (block.block_number * 2)) & (1 << (block.block_number * 2));
+      intmode = CCL_INTMODE0_RISING_gc;
```

Comment:
Why would this be a better solution? More code lines and larger compiled size

Reply:
Because `intmode` field of each block in INTCTRL0 register consists of 2 bits, the original code can't SET intended 2 bit value to the field.

For example, setting FALLING mode after setting RISING on the same block will actually set CHANGE mode into corresponding  2 bits field of INTCTRL0 register. In other case, once CHANGE mode is set, we can't set either RISING and FALLING mode afterwards.